File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n = nums.size();
        
        if(n <= 1) return n;
        
        set<int> hashSet;
        
        // insert elements into hashset
        for(int i = 0; i < n; i++) {
            hashSet.insert(nums[i]);
        }
        
        int longestStreak = 0;
        // traverse the array 
        for(int i = 0; i < n; i++) {
            // in order to count the sequnce we need to know for sure that the element  is the starting element or not 
            // the below conditions checks if the element before nums[i] is not present then only we gonna move further
            if(!hashSet.count(nums[i] - 1)) {
                int currentNum = nums[i];
                int currentStreak = 1;
                
                while(hashSet.count(currentNum + 1)) {
                    currentStreak++;
                    currentNum++;
                }
                
                longestStreak = max(longestStreak, currentStreak);
            }
        }
        
        return longestStreak;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
    hashMap := make(map[int]bool)
    
    // update the hashmap 
    for _, v := range nums {
        hashMap[v] = true
    }
    
    res := 0
    
    // iterate the loop and find the sequence
    for _, v := range nums {
        count := 0;
        // find if this is the sequence starting
        if _, ok := hashMap[v - 1]; ok {
            continue
        } else {
            // if it is a starting sequnce then count the seq
            key := v
            for {
                _, ok := hashMap[key]
                
                if !ok {
                    break
                }
                
                count++
                key++
            }
            
            if count > res {
                res = count
            }
        }
    }
    
    return res
}